lib/bundler/definition.rb

```diff
@@ -184,11 +184,10 @@ def specs_for(groups)
     # @return [SpecSet] resolved dependencies
     def resolve
       @resolve ||= begin
+        last_resolve = converge_locked_specs
```

Comment:
I think you're right. Now that we have sources (and don't save those sources in the lock), we have to resolve the sources using the Gemfile even when it's frozen/locked/etc.